core/src/main/scala/akka/persistence/jdbc/snapshot/dao/AkkaSerializerSnapshotDao.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.snapshot.dao
+
+import slick.jdbc.{ JdbcBackend, JdbcProfile }
+import akka.persistence.SnapshotMetadata
+import akka.persistence.jdbc.config.SnapshotConfig
+import akka.serialization.Serialization
+import akka.stream.Materializer
+import SnapshotTables._
+import akka.dispatch.ExecutionContexts
+import akka.persistence.jdbc.journal.dao.AkkaSerialization
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Success, Try }
+
+class AkkaSerializerSnapshotDao(
+    db: JdbcBackend#Database,
+    profile: JdbcProfile,
+    snapshotConfig: SnapshotConfig,
+    serialization: Serialization)(implicit ec: ExecutionContext, val mat: Materializer)
+    extends SnapshotDao {
+  import profile.api._
+  val queries = new SnapshotQueries(profile, snapshotConfig.snapshotTableConfiguration)
+
+  private def toSnapshotData(row: SnapshotRow): Try[(SnapshotMetadata, Any)] = {
+    val snapshot = serialization.deserialize(row.snapshotPayload, row.snapshotSerId, row.snapshotSerManifest)
+
+    snapshot.flatMap { snapshot =>
+      val metadata = for {
+        mPayload <- row.metaPayload
+        mSerId <- row.metaSerId
+      } yield (mPayload, mSerId)
+
+      metadata match {
+        case None =>
+          Success((SnapshotMetadata(row.persistenceId, row.sequenceNumber, row.created), snapshot))
+        case Some((payload, id)) =>
+          serialization.deserialize(payload, id, row.metaSerManifest.getOrElse("")).map { meta =>
+            (SnapshotMetadata(row.persistenceId, row.sequenceNumber, row.created, Some(meta)), snapshot)
+          }
+      }
+    }
+  }
+
+  private def serializeSnapshot(meta: SnapshotMetadata, snapshot: Any): Try[SnapshotRow] = {
+    val serializedMetadata = meta.metadata.flatMap(m => AkkaSerialization.serialize(serialization, m).toOption)
+    AkkaSerialization
+      .serialize(serialization, payload = snapshot)
+      .map(
+        serializedSnapshot =>
+          SnapshotRow(
+            meta.persistenceId,
+            meta.sequenceNr,
+            meta.timestamp,
+            serializedSnapshot.serId,
+            serializedSnapshot.serManifest,
+            serializedSnapshot.payload,
+            serializedMetadata.map(_.serId),
+            serializedMetadata.map(_.serManifest),
+            serializedMetadata.map(_.payload)))
+  }
+
+  private def zeroOrOneSnapshot(rows: Seq[SnapshotRow]): Option[(SnapshotMetadata, Any)] =
+    // TODO maybe throw instead of toOption otherwise serialization errors will not be logged
```

Comment:
this TODO is important to solve before we merge this one

I agree, we should not call `toOption`